rsciio/arina/_api.py

```diff
@@ -0,0 +1,272 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2025 The HyperSpy developers
+#
+# This library is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with any project and source this library is coupled.
+# If not, see <https://www.gnu.org/licenses/#GPL>.
+
+import logging
+from pathlib import Path
+
+import h5py
+import hdf5plugin  # noqa F401
+import numpy as np
+
+from rsciio._docstrings import (
+    FILENAME_DOC,
+    LAZY_DOC,
+    RETURNS_DOC,
+)
+
+_logger = logging.getLogger(__name__)
+
+
+def file_reader(
+    filename,
+    lazy=False,
+    navigation_shape=None,
+    rebin_diffraction=1,
+    dtype=None,
+    flatfield=None,
+):
+    """Read arina 4D-STEM datasets.
+
+    hdf5plugin is needed in addition to h5py due to compression filters used by the detector
+    when writing data.
+
+    Parameters
+    ----------
+    %s
+    %s
+    navigation_shape : tuple or int or None, default = None
+        Specify the shape of the navigation space. If None, assumes square acquisition.
+        A tuple can be specified as (x_scan_dimension, y_scan_dimension), (x_scan_dimension, "auto"),
+        or ("auto", y_scan_dimension). With "auto" the length is inferred from the number of
+        diffraction patterns.  If only an integer is passed, it assumed to be the x_scan_dimension.
+    rebin_diffraction : int, default=1
+        Diffraction space binning factor for bin-on-load.
+    dtype : float, optional
+        Datatype for dataset.
+    flatfield : numpy.ndarray, optional
+        Flatfield for correction factors, converts data to float.
+    %s
```

Comment:
PR02: Unknown parameters {'-------', 'Returns', 'list of dict'}
  PR04: Parameter "Returns" has no type
  PR07: Parameter "Returns" has no description
  PR04: Parameter "-------" has no type
  PR07: Parameter "-------" has no description
  PR04: Parameter "list of dict" has no type
  RT01: No Returns section found  --> You need an extra return to get numpydoc to know this is a different section
```suggestion

    %s
```